core/src/impl/Kokkos_ClockTic.hpp

```diff
@@ -110,15 +110,25 @@ KOKKOS_IMPL_HOST_FUNCTION inline uint64_t clock_tic_host() noexcept {
 
   return ((uint64_t)a) | (((uint64_t)d) << 32);
 
-#elif defined(__powerpc) || defined(__powerpc__) || defined(__powerpc64__) || \
-    defined(__POWERPC__) || defined(__ppc__) || defined(__ppc64__)
+#elif defined(__powerpc64__) || defined(__ppc64__)
 
-  unsigned int cycles = 0;
+  unsigned long cycles = 0;
 
   asm volatile("mftb %0" : "=r"(cycles));
 
   return (uint64_t)cycles;
 
+#elif defined(__powerpc) || defined(__powerpc__) || defined(__POWERPC__) || \
+    defined(__ppc__)
+
+  unsigned int lower_word = 0;
+  unsigned int upper_word = 0;
+
+  asm volatile("mftb %0" : "=r"(lower_word));
+  asm volatile("mftbu %0" : "=r"(upper_word));
+
+  return (((uint64_t)upper_word) << 32 | ((uint64_t)lower_word));
+
```

Comment:
The 32-bit PPC code path is probably completely irrelevant - Kokkos is never expecting to be run on machines like this, since they haven't been used in HPC in probably 15+ years

Reply:
Since we don’t have a 32-bit PPC system to test against, I’d be hesitant to introduce code for it that may have some subtle issue. Given that, I’d be more inclined to just fall back to the `#else` case and not introduce this `#elif`

Reply:
dropped in [363df9](https://github.com/kokkos/kokkos/pull/5577/commits/363df9228444cc6f2636232cfc09a62c0ec2ae5e)